### PR TITLE
Add support for role and cluster role 

### PR DESCRIPTION
**Pull request by kaitoii11**
**What this PR does / why we need it**:
Add support for generating metrics for `clusterroles`  and `roles`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref #1717

### CODE REVIEW

File: docs/clusterrole-metrics.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# ClusterRole Metrics
+
+| Metric name| Metric type | Labels/tags | Status |
+| ---------- | ----------- | ----------- | ----------- |
+| kube_clusterole_annotations | Gauge | `clusterole`=&lt;clusterole-name&gt; | EXPERIMENTAL
```
Comment (mrueg) [inline docs/clusterrole-metrics.md:5]: Check for clusterole in this doc and replace with cluste*r*role

**Review by liangyuanpeng — APPROVED** [review]

**Comment by mrueg** [comment]
@kaitoii11 can you rebase on latest default branch and run regenerate manifests via make locally?

Thanks!

**Review by fpetkovski — COMMENTED** [review]
/lgtm 👍

**Pull request merged**